### ISSUE TITLE
socket hang up when using node-inspector

### ISSUE THREAD

**Issue by floydwch** [typicode/hotel#59]
Socket hang up when using node-inspector to pause the execution, to be more exact, the socket hang up when the execution is continued.

```
node_modules/hotel/node_modules/http-proxy/lib/http-proxy/index.js:119
    throw err;
    ^

Error: socket hang up
    at createHangUpError (_http_client.js:239:15)
    at Socket.socketCloseListener (_http_client.js:271:23)
    at emitOne (events.js:95:20)
    at Socket.emit (events.js:182:7)
    at TCP._onclose (net.js:477:12)
```

**Comment by floydwch** [comment]
This issue seems to have the same root cause with this PR:
https://github.com/typicode/hotel/pull/61

**Comment by typicode** [comment]
Hey, thank you for the PR and link to http-proxy issue :+1: 
By the way, I'm not familiar with `node-inspector` how could I reproduce the error?

**Comment by floydwch** [comment]
@typicode Install `node-inspector` via `npm i -g node-inspector`, then start the nodejs app with `--debug` to enable node's debug mode, i.e. `node --debug app.js`. Node will listen to the port 5858 to wait for  `node-inspector`, then run `node-inspector`.

 `node-inspector` will listen to the port 8080 and take a parameter to specify the node app's debug port, i.e. `http://127.0.0.1:8080/?port=5858`. Visit the url then set a breakpoint on the app's source code from the panel.

Finally, trigger the breakpoint from the node app's page i.e. go to `http://app.dev`, and it'll pause on the breakpoint then resume its execution, and you'll encounter the issue.

**Comment by typicode** [comment]
Awesome, thanks :)

**Comment by typicode** [comment]
Fix released in `v0.4.9`
